### ISSUE TITLE
bug(false-positive): TypeScript arrow function return type

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#910]
code:

```ts
export const makeTemplate =
  (str: string): ((values?: Array<unknown>) => string) =>
  (values = []) =>
    str.replace(/\$(\d+)/g, (_match, number) => {
      if (!Array.isArray(values)) {
        throw new Error('Array of values must be passed to the template.');
      }
      return values[number - 1];
    });
```

expected: no squigly

actual:
![image](https://user-images.githubusercontent.com/5297725/212584416-f2f69539-90df-41bc-b258-5b46dfd6033f.png)

note: this works

```ts
export const makeTemplate =
  (str: string): (values?: Array<unknown>) => string =>
  (values = []) =>
    str.replace(/\$(\d+)/g, (_match, number) => {
      if (!Array.isArray(values)) {
        throw new Error('Array of values must be passed to the template.');
      }
      return values[number - 1];
    });
```

**Comment by strager** [comment]
Fixed in commit 0c41888aaa573fd3812502e4b7601c93a09e029a. Thanks for reporting!
